src/Pkg.jl

```diff
@@ -41,14 +41,18 @@ const RESPECT_SYSIMAGE_VERSIONS = Ref(true)
 # For globally overriding in e.g. tests
 const DEFAULT_IO = Ref{Union{IO,Nothing}}(nothing)
 
-struct UnstableIO <: IO
-    io::IO
+# See discussion in https://github.com/JuliaLang/julia/pull/52249
+const UnstableIO = IOContext{IO}
+function unstableio(@nospecialize(io::IO))
+    # Needed to prevent specialization https://github.com/JuliaLang/julia/pull/52249#discussion_r1401199265
+    _io = Base.inferencebarrier(io)
+    IOContext{IO}(
+        _io,
+        get(_io,:color,false) ? Base.ImmutableDict{Symbol,Any}(:color, true) : Base.ImmutableDict{Symbol,Any}()
```

Comment:
Why is this needed?

Reply:
Do you have another suggestion as how one might construct a `IOContext{IO}`?

```julia
julia> IOContext{IO}(stdout)
ERROR: MethodError: no method matching IOContext{IO}(::Base.TTY)

Closest candidates are:
  IOContext{IO_t}(::IO_t, ::Base.ImmutableDict{Symbol, Any}) where IO_t<:IO
   @ Base show.jl:291
```

Reply:
An alternative is that I use `Base.unwrapcontext` instead of copying the code. I'm not sure if that complicates the inference barrier or not.

Reply:
This does cause compilation of `get` for the `IO` though.